Software reference update for v0.28/ v1.0.0 (#759)
astronomer/docs: software_versioned_docs/version-0.28/cli-reference.md

### Confirm the install

> To make sure that you have the Astro CLI installed on your machine, run:

Reviewer: ```suggestion
To make sure that you have the Astro CLI installed on your computer, run:
```

The settings that you can update via the command line are:

- `cloud.api.protocol`
- `cloud.api.port`
- `cloud.api.ws_protocol`
- `cloud.api.token`
- `container.engine`
- `context`
- `contexts`
- `houston.dial_timeout`
- `local.houston`
- `local.orbit`
- `postgres.user`
- `postgres.password`
- `postgres.host`
- `postgres.port`
- `project.deployment`
- `project.name`
- `project.workspace`
- `webserver.port`
- `show_warnings`

### Subcommands

| Subcommand | Usage                                                     |
| ---------- | --------------------------------------------------------- |
| `get`      | Show current values for the above configuration settings. |
| `set`      | Updates a setting in your platform to a new value.        |

### Related documentation

- [Apply a Platform Configuration Change on Astronomer](apply-platform-config.md)

## astro context delete

> Delete the locally stored information for a given Astronomer installation. After running this command, the domain for the installation that you specify will no longer appear when you run `astro context list`, and you will not be able to switch to the installation via `astro context switch`.

Reviewer: ```suggestion
Delete the locally stored information for a given Astronomer installation. After running this command, the domain for the installation that you specify no longer appears when you run `astro context list`, and you will not be able to switch to the installation via `astro context switch`.
```

> If you re-authenticate to an installation that you previously deleted via this command, its information will again be available from `astro context list` and `astro context switch`.

Reviewer: ```suggestion
If you use this command to reauthenticate to an installation that you previously deleted, installation information is available when you use the `astro context list` and `astro context switch` commands.
```

### Usage

```sh
astro context delete <basedomain>
```

## astro context list

> View a list of domains for all Astronomer installations that you have access to. An Astronomer installation will appear on this list if you have authenticated to it at least once using `astro login`.

Reviewer: ```suggestion
View a list of domains for all Astronomer installations that you have access to. Astronomer installations appear on this list if you have authenticated to it at least once using `astro login`.
```

### Usage

```sh
astro context list
```

## astro context switch

Switch to a different Astronomer installation. You can switch to a given Astronomer installation if you have authenticated to it at least once using `astro login`.

> Note that after switching to a different Astronomer installation, you might have to re-authenticate to the installation using `astro login`.

Reviewer: ```suggestion
After you switch to a different Astronomer installation, you might need to run `astro login` to reauthenticate to the installation.
```

Run `astro deploy <your-deployment-release-name> [flags]` in your terminal to push a local Airflow project as a Docker image to your Airflow Deployment on Astronomer.

If you have the appropriate Workspace and Deployment-level permissions, your code is packaged into a Docker image, pushed to Astronomer's Docker Registry, and applied to your Airflow Webserver, Scheduler(s), and Worker(s). Before completing the process, it tests your DAGs in your Astro project for errors. If this test fails, the deploy to Astro will also fail. This is the same test which runs locally with `astro dev parse`.

To identify your Deployment's release name, go to **Settings** > **Basics** > **Release Name** in the Software UI or run `astro deployment list`.

> If you run `astro deploy` without specifying `your-deployment-release-name`, the Astro CLI will list all Airflow Deployments in your Workspace to choose from.

Reviewer: ```suggestion
If you run `astro deploy` without specifying `your-deployment-release-name`, the Astro CLI lists all Airflow Deployments in your Workspace.
```

Run `astro deployment airflow upgrade --deployment-id` to initialize the Airflow upgrade process. To finalize the Airflow upgrade process, complete all of the steps as described in [Upgrade Apache Airflow on Astronomer](manage-airflow-versions.md).

> If you do not specify `--desired-airflow-version`, this command will output a list of available versions of Airflow you can choose from and prompt you to pick one. The Astro CLI will only make available versions of Airflow that are higher than the version you're currently running in your `Dockerfile`.

Reviewer: ```suggestion
If you do not specify `--desired-airflow-version`, this command creates a list of available Airflow versions that you can select. The Astro CLI lists only the available Airflow versions that are later than the version currently specified in your `Dockerfile`.
```

### Flags

| Flag                    | Value Type | Usage                                                                                                                                                                                                         |
| ----------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--airflow-version`     | String     | The Airflow version for the new Deployment.                                                                                                                                                                   |

> | `--cloud-role`          | String     | Append an AWS or GCP IAM role to your Airflow Deployment's Webserver, Scheduler, and Worker Pods.                                                                                                             |

Reviewer: ```suggestion
| `--cloud-role`          | String     | Append an AWS or GCP IAM role to your Airflow Deployment Webserver, Scheduler, and Worker Pods.                                                                                                             |
```

### Flags

| Flag                         | Value Type | Usage                                                                                                                           |
| ---------------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------- |
| `--category`                 | String     | The category for the new service account as displayed in the Software UI. This is optional, and the default value is `Not set`. |
| `--deployment-id` (Required) | String     | The Deployment you're creating a service account for.                                                                           |
| `--label` (Required)         | String     | The name or label for the new service account.                                                                                  |

> | `--role`                     | String     | The User Role for the new service account. Can be `viewer`, `editor`, or `admin`. The default value is `viewer`.                |

Reviewer: ```suggestion
| `--role`                     | String     | The User Role for the new service account. Select `viewer`, `editor`, or `admin`. The default is `viewer`.                |
```

| `--dag-deployment-type` | String     | The DAG deploy method for the Deployment. Can be either `image` or `volume`. The default value is `image`.                                                                                                    |
| `--nfs-location`        | String     | The location for an NFS volume mount, specified as: `<IP>:/<path>`. Must be specified when `--dag-deployment-type=volume`. Input is automatically prepended with `nfs:/` - do not include this in your input. |
| `label`                 | String     | The label or name for the Deployment.                                                                                                                                                                                 |
| `description`           | String     | The description for a Deployment.                                                                                                                                                                             |

> | `version`               | String     | The Airflow version for the Deployment (e.g. `v2.0.0`).                                                                                                                                                       |

Reviewer: ```suggestion
| `version`               | String     | The Airflow version for the Deployment (foe example, `v2.0.0`).                                                                                                                                                       |
```

| `releaseName`           | String     | The release name for the Deployment (e.g. `planetary-fusion-1382`).                                                                                                                                           |

> | `alert_emails`          | String     | An email address which receives Airflow alerts from the Deployment.                                                                                                                                           |

Reviewer: ```suggestion
| `alert_emails`          | String     | The email address for Airflow alerts sent from the Deployment.                                                                                                                                           |
```

### Usage

`astro deployment user add --email=<user-email-address> --deployment-id=<user-deployment-id> --role<user-role>`

### Flags

| Flag                         | Value Type | Usage                                                                                                                                            |
| ---------------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |

> | `--deployment-id` (Required) | String     | The ID of the Deployment that the user will be added to. To find this value, run `astro deployment list`.                                        |

Reviewer: ```suggestion
| `--deployment-id` (Required) | String     | The ID for the Deployment that the user is added to. To find this value, run `astro deployment list`.                                        |
```

## astro dev init

Initializes a new Astro project in your working directory. The set of files generated by this command are required to run Airflow locally and can be pushed to a Deployment on Astronomer.

:::info

> To deploy your project to a Software Deployment, you must either specify the `--use-astronomer-certified` flag when running `astro dev init` or update the image in your project's `Dockerfile` to a supported Astronomer Certified image. For more information, see [Deploy DAGs via Astro CLI](deploy-cli.md).

Reviewer: ```suggestion
To deploy your project to a Software Deployment, you must specify the `--use-astronomer-certified` flag when running `astro dev init`, or update the image in your project's `Dockerfile` to a supported Astronomer Certified image. For more information, see [Deploy DAGs via Astro CLI](deploy-cli.md).
```

## astro dev pytest

Run unit tests for your data pipelines with `pytest`, a testing framework for Python. When you run this command, the Astro CLI creates a local Python environment that includes your DAG code, dependencies, and Astronomer Certified Docker image. The CLI then runs any pytests in the `tests` directory of your Astro project and shows you the results of those tests in your terminal.

For more information on this functionality, see [Test and Troubleshoot Locally](test-and-troubleshoot-locally.md).

### Usage

`astro dev pytest` to run specfic Pytests on your DAGs. Use `astro dev pytest <pytest-filepath>` to specify a specific test.

### Options

| Flag          | Value Type | Usage                                              |
| ------------- | ---------- | -------------------------------------------------- |
|`<pytest-filepath>`| String | Any valid filepath within the `tests` directory. |
| `--env`       | string     | The filepath to your environment variables. (The default is `.env`)  |

## astro dev restart

Stop your Airflow environment, rebuild your Astro project into a Docker image, and restart your Airflow environment with the new Docker image.

> This command can be used to rebuild an Astro project and run it locally. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).

Reviewer: ```suggestion
You can use this command to rebuild an Astro project and run it locally. See [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).
```

- [Running the Airflow Upgrade Check Package](https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html#upgrade-check)

## astro login/ logout

Logs you in and out of an installation on Astronomer Software.

### Usage

Run `astro login <base-domain>` or `astro logout <base-domain>` to log in or out of your Astronomer platform respectively. This is equivalent to using the login screen of the Software UI.

> If you have access to more than one Astronomer installation, each will have a unique `<base-domain>`. When switching between platforms, make sure to log out of one `<base domain>` before logging into another.

Reviewer: ```suggestion
If you have access to more than one Astronomer installation, each installation has a unique `<base-domain>`. When moving between platforms, make sure to log out of one `<base domain>` before logging into another.
```

## astro version

> Displays the running versions of both the Astro CLI and the Astronomer platform to which you are authenticated. If the minor versions of the Astro CLI and your Astronomer platform don't match, we encourage you to upgrade.

Reviewer: ```suggestion
Displays the running versions of both the Astro CLI and the Astronomer platform to which you are authenticated. Astronomer recommends upgrading when the minor versions of the Astro CLI and your Astronomer platform don't match.
```

### Flags

| Flag                        | Value Type | Usage                                                                                                                                                |
| --------------------------- | ---------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--workspace-id` (Required) | String     | The Workspace you're creating a service account for.                                                                                                 |
| `--label` (Required)        | String     | A label for the service account.                                                                                                                     |

> | `--category`                | String     | The Category for the service account. The default value is `Not set`.                                                                                |

Reviewer: ```suggestion
| `--category`                | String     | The Category for the service account. The default is `Not set`.                                                                                |
```

### Usage

`astro workspace user add --email <user-email-address> [flags]`

### Flags

| Flag                   | Value Type | Usage                                                                                                                                                                     |
| ---------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--email` (_Required_) | String     | The user's email.                                                                                                                                                         |

> | `--workspace-id`       | String     | The Workspace that the user will be added to. Specify this flag if you want to create a user in a Workspace other than your current Workspace.                            |

Reviewer: ```suggestion
| `--workspace-id`       | String     | The Workspace that the user is added to. Specify this flag if you want to create a user in a Workspace that is different than your current Workspace.                            |
```

> | `--workspace-id` | String     | The Workspace that you're searching in. Specify this flag if you want to search for users in a Workspace other than your current Workspace. |

Reviewer: ```suggestion
| `--workspace-id` | String     | The Workspace that you're searching in. Specify this flag if you want to search for users in a Workspace that is different than your current Workspace. |
```
